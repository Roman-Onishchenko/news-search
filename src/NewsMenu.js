import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from './actions';
import { Link } from 'react-router-dom';
import './styles/NewsList.css';

export class NewsMenu extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (
    <div className="news-list">
      <ul>
        <li><Link to="/bbc-sport">Спортивные новости</Link></li>
        <li><Link to="/the-economist">Финансовые новости</Link></li>
        <li><Link to="/techradar">Новости науки</Link></li>
        <li><Link to="/usa-today">Зарубежные новости</Link></li>
        <li><Link to="/google-news">Новости культуры</Link></li>
        <li><Link to="/fortune">Разное</Link></li>
      </ul>
    </div>
  )}
}

export default connect(
  state => ({

  }),
  dispatch => bindActionCreators(actions, dispatch)
)(NewsMenu);
