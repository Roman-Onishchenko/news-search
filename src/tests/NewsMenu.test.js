import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import sinon from 'sinon';
import { Route, Link, MemoryRouter } from 'react-router-dom';
import { NewsMenu } from '../NewsMenu';
import toJson from 'enzyme-to-json';

describe('NewsMenu', () => {

  it('renders without crashing', () => {

    const renderedComponent = shallow(
        <NewsMenu />
    )
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

});
