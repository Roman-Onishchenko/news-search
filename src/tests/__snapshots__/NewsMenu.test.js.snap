// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsMenu renders without crashing 1`] = `
<div
  className="news-list"
>
  <ul>
    <li>
      <Link
        replace={false}
        to="/bbc-sport"
      >
        Спортивные новости
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/the-economist"
      >
        Финансовые новости
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/techradar"
      >
        Новости науки
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/usa-today"
      >
        Зарубежные новости
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/google-news"
      >
        Новости культуры
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/fortune"
      >
        Разное
      </Link>
    </li>
  </ul>
</div>
`;
